public/js/util/source-map.js

```diff
@@ -1,29 +1,111 @@
 "use strict";
 
 const { networkRequest } = require("./networkRequest");
-const { SourceMapConsumer } = require("source-map");
+const invariant = require("invariant");
+const { SourceMapConsumer, SourceNode } = require("source-map");
+const { Source } = require("../types");
+const { Task } = require("./task");
+const { getSource } = require("../selectors");
 
-let sourceMaps = new Map();
+let sourceMapConsumers = new Map();
+let sourceNodes = new Map();
+
+function hasConsumer(sourceId) {
+  return sourceMapConsumers.has(sourceId);
+}
+
+function getConsumer(sourceId) {
+  return sourceMapConsumers.get(sourceId);
+}
+
+function getOriginalSources(sourceId) {
+  const consumer = getConsumer(sourceId);
+  return consumer.sources;
+}
+
+function isOriginal(originalSource) {
+  return !!getGeneratedSourceId(originalSource);
```

Comment:
It might be better to come up with a sourceId format that we can simply match here. That way we don't have to iterate all the consumers and all the sources in each of them every time you click on a source.